test/e2e/nftables_test.go

```diff
@@ -25,28 +26,35 @@ var _ = Describe("Nftables", func() {
 	It("should apply firewall by blocking all ports except the ones OCP is listening on", func() {
 		masterMat, workerMat := commatrix.SeparateMatrixByRole()
 		var workerNFT []byte
-
+		nodeRoles := []string{"master"}
 		By("Creating NFT output for each role")
 		masterNFT, err := masterMat.ToNFTables()
 		Expect(err).NotTo(HaveOccurred())
 		if !isSNO {
+			nodeRoles = append(nodeRoles, "worker")
 			workerNFT, err = workerMat.ToNFTables()
 			Expect(err).NotTo(HaveOccurred())
 		}
-
 		g := new(errgroup.Group)
-		for _, node := range nodeList.Items {
-			nodeName := node.Name
-			nodeRole, err := types.GetNodeRole(&node)
-			Expect(err).ToNot(HaveOccurred())
+		versionMajorMinor, err := utilsHelpers.GetClusterVersion()
+		Expect(err).ToNot(HaveOccurred())
+
+		if firewall.IsVersionGreaterThan(versionMajorMinor, "4.16") { // if version more than 4.16 need to change cluster MachineConfiguration.
```

Comment:
The communication matrix is only supported since 4.16

Reply:
Yah yah
The meaning is from 4.17+
And for 4.16 we are not doing that